framework/framework.go

```diff
@@ -105,12 +105,12 @@ func (f *Framework) Boot() {
 		}
 
 		notifier := func(err error, d time.Duration) {
-			f.logger.LogCtx(ctx, "warning", fmt.Sprintf("retrying operator boot due to error: %#v", microerror.Mask(err)))
+			f.logger.LogCtx(ctx, "function", "Boot", "level", "warning", "message", "retrying framework boot due to error", "stack", fmt.Sprintf("%#v", err))
```

Comment:
This is about some more structure in our logs. We use structured logging so we should more properly make use of it. One common term we are interested in in different places is `function`. Then there is the log level which we used implicitly but I want to have this to be used explicitly and therefore there is now the key `level`. The message is now more clear and clean. For errors there is a key `stack` which is then also more consistent across the code base.

Reply:
I see an improvement there, but are you going to change all logs everywhere?

BTW framework.go isn't a good place to start. Please better check `crud_resource_ops.go` as most of that stuff will be wiped out.

Reply:
As I said, I am glad to continue in other places over time.

Reply:
OK, when you think it's worth it. Did you think about just adding methods to logger? Like `.Debug`, `.Info` and so on?

Also when we change the logging style I think it would be nice to have more ppl involved in the discussion.

Reply:
I do not want to extend the logger interface. Towards getting more people on board, we can do this. I just thought I can be productive and boyscout without a lot of trouble and waiting would just kill a bit of my motivation now.

Reply:
I've heard about changing approach to logging a few times already. Probably guys have their preferences, as this touches more (all) projects than operatorkit. For me it doesn't really make a difference, but I don't want to actively (passively is ok) replace logs we have now. I'm happy you want to do it eventually. Killing motivation is not my intention, so LGTM.

Reply:
All good. I just did not think this is a big issue everybody is concerned about so I thought I can go ahead more easily. I will wait for additional feedback now. If there does not happen anything, which I suspect, I will go ahead.

Reply:
Like @kopiczko I'm against actively changing this because it causes too much code churn for the value provided. Changing passively is fine with me.

Reply:
I am not sure what this means. What is passive or active change here?

Reply:
To me passive means we change as we create or modify components. Active would be there is a task to update all components.

Reply:
I understand it like that. When I write a new log I use the new approach. When I update a log I adjust it to the new approach. But I don't want to go and fix all logs in a project or something like that.

Reply:
Ok cool. I think we are on the same page. There might be a time where I want to go through an operator or package and just change it in a separate PR. I do not force people to do this though.